### ISSUE TITLE
cql-pytest test_secondary_index.py::test_order_of_indexes warns of deprecation

### ISSUE THREAD

**Issue by nyh** [scylladb/scylladb#8752]
If we run `test/cql-pytest/run `**`test_secondary_index.py::test_order_of_indexes`** (or all tests), we get a deprecation warning:
```
=============================== warnings summary ===============================
test_secondary_index.py::test_order_of_indexes
test_secondary_index.py::test_order_of_indexes
test_secondary_index.py::test_order_of_indexes
test_secondary_index.py::test_order_of_indexes
test_secondary_index.py::test_order_of_indexes
test_secondary_index.py::test_order_of_indexes
  /home/nyh/scylla.maintainer/test/cql-pytest/test_secondary_index.py:92: DeprecationWarning: ResultSet indexing support will be removed in 4.0. Consider using ResultSet.one() to get a single row.
    assert any([index_name in event.description for event in cql.execute(query, trace=True).get_query_trace().events])
```

As this message says, the problem is the call to
```
cql.execute(query, trace=True).get_query_trace()
```
It's not even clear to which of the possibly many pages of results that `execute()` may need to request this trace relates to. The deprecation warning suggests to use
```
cql.execute(query, trace=True).one().get_query_trace()
```
But this doesn't actually seem to work - in my tests, the query returns an empty result set, where one() returns None, and `get_query_trace()` can't be called on that. Maybe instead we need to loop over all (zero or more) the responses of execute and do `get_query_trace()` on them? BUT, if the result set is empty in every query in this test, it won't check anything?!

@psarna can you please take a look at this - unfortunately I don't understand how this test is supposed to work, and since it's a failing test that doesn't work on Cassandra, I don't know what to expect the results to be.

This is a low-priority issue because it's annoying to see this warning every time, but it can't cause a test failure because the test is already marked "xfail" so if it starts failing instead of warning, we won't care.

**Comment by psarna** [comment]
@nyh That's actually a bug in the driver - it internally uses a mechanism it itself marked as deprecated when populating tracing results. Here's a pull request for the original driver I just sent: https://github.com/datastax/python-driver/pull/1103 ; if it doesn't go in for any reason, we can just independently apply it on our shard-aware fork

**Comment by nyh** [comment]
Thanks! Let's leave this issue open until they fix this driver bug, so if we encounter these warnings again, we can find the open issue.

**Comment by mykaul** [comment]
> Thanks! Let's leave this issue open until they fix this driver bug, so if we encounter these warnings again, we can find the open issue.

The driver is fixed, I hope the issue is gone now.

**Comment by DoronArazii** [comment]
Closing as completed.
Let's re-open if necessary.

**Comment by nyh** [comment]
Sadly, it turns out I made a mistake, in commit 0a71151bc4aed36850e04ac4c3a890fe00e05630 I got rid of the warning mentioned in this issue by doing `cql.execute(query, trace=True).one().get_query_trace()` - **exactly** the command which I noted above doesn't work. And I didn't notice that it doesn't work because the test was xfailing anyway :-(

So the reason why this test stopped printing warnings was that it changed (0a71151bc4aed36850e04ac4c3a890fe00e05630), not a driver fix.

I'll check now if after reverting the wrong 0a71151bc4aed36850e04ac4c3a890fe00e05630 this issue is still solved (by virtue of the driver improvement) and only then can close this issue.
